extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/multipart/MultipartMessageBodyWriter.java

```diff
@@ -0,0 +1,165 @@
+package io.quarkus.resteasy.reactive.server.runtime.multipart;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import javax.ws.rs.RuntimeType;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.resteasy.reactive.common.core.Serialisers;
+import org.jboss.resteasy.reactive.common.reflection.ReflectionBeanFactoryCreator;
+import org.jboss.resteasy.reactive.server.core.CurrentRequestManager;
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.core.ServerSerialisers;
+import org.jboss.resteasy.reactive.server.spi.ServerMessageBodyWriter;
+import org.jboss.resteasy.reactive.server.spi.ServerRequestContext;
+import org.jboss.resteasy.reactive.spi.BeanFactory;
+
+public class MultipartMessageBodyWriter extends ServerMessageBodyWriter.AllWriteableMessageBodyWriter {
+
+    private static final String RESTEASY_DEFAULT_CHARSET_PROPERTY = "quarkus.resteasy-reactive.multipart.input-part.default-charset";
+    private static final Charset RESTEASY_DEFAULT_CHARSET_PROPERTY_DEFAULT = StandardCharsets.UTF_8;
+    private static final String DOUBLE_DASH = "--";
+    private static final String LINE_SEPARATOR = "\r\n";
+    private static final String BOUNDARY_PARAM = "boundary";
+
+    private volatile Charset defaultCharset;
```

Comment:
I guess I was not clear enough.

`volatile` is only needed if the field is **not** set in the constructor, but in a method that can be called by other threads.

If you set the field in the constructor, `volatile` is not needed and the field can be `final`.

Reply:
Ok, it makes more sense now. Changed.